pkg/authconfig/authconfig_handler.go

```diff
@@ -0,0 +1,53 @@
+package authconfig
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+
+	"github.com/codeready-toolchain/registration-service/pkg/configuration"
+	"github.com/gin-gonic/gin"
+)
+
+// Service implements the auth config endpoint, which is invoked to 
+// retrieve the auth config for the ui.
+type Service struct {
+	config *configuration.Registry
+	logger *log.Logger
+}
+
+// New returns a new Service instance.
+func New(logger *log.Logger, config *configuration.Registry) *Service {
+	return &Service{
+		logger: logger,
+		config: config,
+	}
+}
+
+func (srv *Service) isJSON(s string) bool {
+	var elements map[string]interface{}
+	err := json.Unmarshal([]byte(s), &elements)
+	if err!=nil {
+		srv.logger.Println("error parsing auth config JSON:", err.Error())
+		return false
+	}
+	return true
+}
+
+// AuthconfigHandler returns signup info.
+func (srv *Service) AuthconfigHandler(ctx *gin.Context) {
+	ctx.Writer.Header().Set("Content-Type", "application/json")
+	ctx.Writer.WriteHeader(http.StatusOK)
+	configStr := srv.config.GetAuthClientConfigAuthJSON()
+ 	if !srv.isJSON(configStr) {
```

Comment:
Do we really care if it's JSON or not? We are returning here the raw config which is used only by front-end and which we just host in the backend as is, right? So, shouldn't we just return this raw string in the body here?
If we want to do some kind of validation of the config value then this should be done in the configuration package when we are loading the configuration. But not in the `/authconfig` handler every time this endpoint is called.

There should be some JIRA for improving our configuration loading. We need to add config validation when initializing our configuration and fail our health check if there is something wrong with the configuration, so we don't start a service in prod with invalid configuration. If we don't have such JIRA we should create it.

Reply:
I need to check, but I intentionally put the validation into the handler as I'd expect config changes be recognized on change, not on restart. But on the other hand, it's a good point that the config should not have any req's attached. I'll remove that check,

Reply:
Created a story for the config check: https://jira.coreos.com/browse/CRT-255

Reply:
Fixed 9259857. Also renamed the config key from ..json to ..raw.

Reply:
Also added a `content-type` config value in bc604a2.

Reply:
> I need to check, but I intentionally put the validation into the handler as I'd expect config changes be recognized on change, not on restart.

Unfortunately we can't do that.
We will use env vars with mapping to a ConfigMap in production to pass the configuration to viper/configuration. So, a pod/service restart is required to pick something new from a ConfigMap -> env var -> viper.
Also I remember viper had some race condition issues if we try to modify the underlying map of already-in-use viper instance. But even if it's not thread safe we still can't use it because of how ConfgiMap - env var mapping works in kube/openshift.

Reply:
@alexeykazakov @michaelkleinhenz Actually, it should work now: Viper can now detect changes in Config Maps and update the config accordingly. I fixed it in https://github.com/spf13/viper/pull/415. It's worth the try to verify, though ;)